src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -118,11 +118,27 @@ abstract class LazyModule()(implicit val p: Parameters)
     children.foreach( _.childrenIterator(iterfunc) )
   }
 
+  def childrenFinder(filter: LazyModule => Boolean): LazyModule = {
+    val found = scala.collection.mutable.ListBuffer[LazyModule]()
+    childrenIterator(lm => if (filter(lm)) found += lm)
+    require(found.nonEmpty, s"no LazyModule found")
+    require(found.size == 1, s"constraint of filter is too loose, found ${found.size} LazyModules")
+    found.head
+  }
+
   def nodeIterator(iterfunc: (BaseNode) => Unit): Unit = {
     nodes.foreach(iterfunc)
     childrenIterator(_.nodes.foreach(iterfunc))
   }
 
+  def nodeFinder(filter: BaseNode => Boolean): BaseNode = {
```

Comment:
should this be called "findNode"?

Reply:
i think it would have to be `findExactlyOneNode`